### PR TITLE
Coverity fixes 4

### PR DESCRIPTION
**Pull request by jan-cerny**
This pull request fixes some of new issues reported by Coverity. See commit messages.

### CODE REVIEW

File: src/XCCDF/xccdf_session.c [hunk 1]
```diff
@@ -486,8 +486,7 @@ int xccdf_session_set_profile_id_by_suffix(struct xccdf_session *session, const
 		return 2;
 	} else {
 		const bool search_result = xccdf_session_set_profile_id(session, full_profile_id);
-		assert(search_result);
-		return 0;
+		return (search_result ? 0 : 1);
```
Comment (mpreisler) [inline src/XCCDF/xccdf_session.c]: this variable is superfluous, why not just use the result of set_profile_id directly?

File: src/common/list.c [hunk 2]
```diff
@@ -393,7 +393,7 @@ struct oscap_htable *oscap_htable_new1(oscap_compare_func cmp, size_t hsize)
 		return NULL;
 	t->hsize = hsize;
 	t->itemcount = 0;
-	t->table = calloc(hsize, sizeof(struct oscap_list_item *));
+	t->table = calloc(hsize, sizeof(struct oscap_htable_item *));
```
Comment (mpreisler) [inline src/common/list.c]: wow, just wow

:D

File: src/XCCDF/xccdf_session.c (hunk 1 above)
Comment (jan-cerny) [reply src/XCCDF/xccdf_session.c]: ok

File: src/common/list.c (hunk 2 above)
Comment (jan-cerny) [reply src/common/list.c]: It didn't cause problems because size of pointer to a structure is same as size of pointer to a different structure.

**Comment by jan-cerny** [comment]
I have amended the first commit so that there is no superfulous variable anymore.

**Pull request merged**